test/remote/gateways/remote_xpay_test.rb

```diff
@@ -2,14 +2,47 @@
 
 class RemoteRapydTest < Test::Unit::TestCase
   def setup
-    @gateway = XpayGateway.new(fixtures(:x_pay))
-    @amount = 200
-    @credit_card = credit_card('4111111111111111')
-    @options = {}
+    @gateway = XpayGateway.new(fixtures(:xpay))
+    @amount = 100
+    @credit_card = credit_card(
+      '4349940199004549',
+      month: 5,
+      year: 2026,
+      first_name: 'John',
+      last_name: 'Smith',
+      verification_value: '396',
+      brand: 'visa'
+    )
+
+    @options = {
+      order_id: SecureRandom.alphanumeric(10),
+      order: {
+        currency: 'EUR',
+        amount: @amount,
+        customer_info: {
+          card_holder_name: 'Ryan Reynolds',
+          card_holder_email: nil,
+          billing_address: address
+        }
+      }
+    }
   end
 
   def test_successful_purchase
     response = @gateway.purchase(@amount, @credit_card, @options)
-    assert response
+    assert_success response
+    assert_match 'PENDING', response.message
+  end
+
+  def test_failed_purchase
```

Comment:
💬 Comment:

The only thing that we can control is the initial request, may be can at least check if the 3DS ACS URL and the other params are present ?

Reply:
Noted, additional asserts added!